safe-global/web-core
# feat: relay a transaction trough the `sponsoredCall` endpoint

## What it solves

Partially resolves https://github.com/safe-global/web-core/issues/1700

## How this PR fixes it
Adds a function to call the tx relayer if the relaying feature is enabled in the current chain and if the connected wallet can relay the tx. SC wallets can only relay fully signed transactions.
Adds a Relaying event to the observable events.
Monitors the relay transaction. The monitor timeout after 3 minutes.

## How to test it

_1/N Safe_
1. Create a transaction.
2. Check the "Execute transaction".
3. Opt for "Via relayer" execution method.
4. Submit

_m/N Safe_
1. Click to sign a transaction that can you can execute.
2. Check the "Execute transaction".
3. Opt for "Via relayer" execution method.
4. Submit

5. Repeat previous steps (1 - 4) with a SC wallet connected.
6. The execution method option shall not appear.

_Fully signed transaction_
1. Click to execute a fully signed transaction.
2. Opt for "Via relayer" execution method.
4. Submit

You shall see the notifications for the Relay transaction as well as the TxStatus changing from "Relaying" -> "indexing" -> "Success"

## Screenshots
![Screenshot 2023-03-06 at 14 38 16](https://user-images.githubusercontent.com/32431609/223125819-79c3580b-5e7a-4f91-85d4-3ebe114f56e0.png)
![Screenshot 2023-03-06 at 14 38 33](https://user-images.githubusercontent.com/32431609/223125816-175b5a12-c83f-446b-abee-dddd2c283f9a.png)


## Checklist
